errtracker/errtracker.go

```diff
@@ -67,6 +69,106 @@ var (
 	timeNow  = time.Now
 )
 
+type reportsDB struct {
+	db *bolt.DB
+
+	// map of hash(dupsig) -> time-of-report
+	reportedBucket *bolt.Bucket
+
+	// time until an error report is cleaned from the database,
+	// usually 7 days
+	cleanupTime time.Duration
+}
+
+func hashString(s string) string {
+	h := sha512.New()
+	io.WriteString(h, s)
+	return fmt.Sprintf("%x", h.Sum(nil))
+}
+
+func newReportsDB(fname string) (*reportsDB, error) {
+	if err := os.MkdirAll(filepath.Dir(fname), 0755); err != nil {
+		return nil, err
+	}
+	bdb, err := bolt.Open(fname, 0600, &bolt.Options{
+		Timeout: 10 * time.Second,
```

Comment:
This timeout is when the database is locked?

Reply:
This should not happen in practise, i.e. this would require two snapds running.